src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/LongPollingTransport.java

```diff
@@ -0,0 +1,116 @@
+        logger.info("Polling {}", pollUrl);
+        HttpRequest request = new HttpRequest();
+        request.addHeaders(headers);
+        return this.pollingClient.get(pollUrl, request).flatMapCompletable(response -> {
+            if (response.getStatusCode() != 200){
+                logger.error("Unexpected response code {}", response.getStatusCode());
+                this.active = false;
+                return Completable.error(new Exception("Failed to connect"));
+            } else {
+                logger.info("Activating poll loop", response.getStatusCode());
+                this.active = true;
+            }
+            poll(url).subscribeWith(pollCompletable);
+
+            return Completable.complete();
+        });
+    }
+
+    private Completable poll(String url){
+        if (this.active) {
+            pollUrl = url + "&_=" + System.currentTimeMillis();
+            logger.info("Polling {}", pollUrl);
+            HttpRequest request = new HttpRequest();
+            request.addHeaders(headers);
+            Completable pollingCompletable = this.pollingClient.get(pollUrl, request).flatMapCompletable(response -> {
+                if (response.getStatusCode() == 204) {
+                    logger.info("LongPolling transport terminated by server.");
+                    this.active = false;
+                } else if (response.getStatusCode() != 200) {
+                    logger.error("Unexpected response code {}", response.getStatusCode());
+                    this.active = false;
+                    this.closeError = "Unexpected response code {}" + response.getStatusCode();
+                } else {
+                    logger.info("Message received");
+                    new Thread(() -> this.onReceive(response.getContent())).start();
+                }
+                return poll(url); });
+            return pollingCompletable;
+        } else {
+            logger.info("Long Polling transport polling complete.");
+            this.onClose.invoke(this.closeError);
+        }
+
+        return Completable.complete();
+    }
+
+    @Override
+    public Completable send(String message) {
+        return Completable.fromSingle(this.client.post(url, message));
+    }
+
+    @Override
+    public void setOnReceive(OnReceiveCallBack callback) {
+        this.onReceiveCallBack = callback;
+    }
+
+    @Override
+    public void onReceive(String message) {
+        this.onReceiveCallBack.invoke(message);
+        logger.debug("OnReceived callback has been invoked.");
+    }
+
+    @Override
+    public void setOnClose(TransportOnClosedCallback onCloseCallback) {
+        this.onClose = onCloseCallback;
+    }
+
+    @Override
+    public Completable stop() {
+        this.active = false;
+        this.pollingClient.delete(this.url);
+        this.pollCompletable.blockingAwait();
+        logger.info("LongPolling transport stopped.");
+        this.onClose.invoke(null);
```

Comment:
Should `onClose` be called both here in stop() and in poll()? Is it OK for it to be called multiple times?

Reply:
The call to onClose in poll is incorrect. Should be stop.